File: FakeLoacationSample/FakeGPS/Category/RuntimeHandler.m
//
//  Swizzle_CLLocationManager.m
//  FakeLoacationSample
//
//  Created by adapter on 2018/11/24.
//  Copyright © 2018 adapter00. All rights reserved.
//

#import "RuntimeHandler.h"

@implementation _RuntimeHandler

+(void)handleLoad {
    NSLog(@"Please override RuntimeHandler.handleLoad if you want to use");
}
+(void)handleInitialize {
    NSLog(@"Please override RuntimeHandler.handleInitialize if you want to use");
}

@end


@implementation RuntimeHandler

+(void)initialize {
    [super initialize];
    [self handleInitialize];
}

+ (void)load {
    [super load];
    [self handleLoad];
}

@end

File: FakeLoacationSample/FakeGPS/RuntimeHandler.swift
//
//  RuntimeHandler.swift
//  FakeLoacationSample
//
//  Created by adapter on 2018/11/25.
//  Copyright © 2018 adapter00. All rights reserved.
//

import Foundation


// This is entrypoint for Method Swizzle
// All subclass of `CLLocationManagerDelegate` is swizzled
extension RuntimeHandler {
    open override class func handleLoad() {
        FakeLocationManager.shared.swizzleCLLocationManagerDelegate()
    }
}
